Library/Homebrew/dev-cmd/determine-cask-runners.rb

```diff
@@ -0,0 +1,351 @@
+
+        conflicts "--url", "--casks"
+        conflicts "--syntax-only", "--url"
+        conflicts "--syntax-only", "--casks"
+        conflicts "--syntax-only", "--skip-install"
+        conflicts "--syntax-only", "--new"
+
+        named_args [:cask], min: 0
+        hide_from_man_page!
+      end
+
+      sig { params(args: T.untyped).void }
+      def initialize(*args)
+        super
+        repository = ENV.fetch("GITHUB_REPOSITORY", nil)
+        raise UsageError, "The GITHUB_REPOSITORY environment variable must be set." if repository.blank?
+
+        @tap = T.let(Tap.fetch(repository), Tap)
+      end
+
+      sig { override.void }
+      def run
+        skip_install = args.skip_install?
+        new_cask = args.new?
+        casks = args.named if args.named.any?
+        pr_url = args.url
+        syntax_only = args.syntax_only?
+        tap = @tap
+
+        raise UsageError, "This command must be run from inside a tap directory." if Dir.pwd != tap.path
+
+        if casks.blank? && pr_url.blank?
+          raise UsageError, "Either a list of cask names or a pull request URL must be provided."
+        end
+
+        labels = if pr_url
+          pr = GitHub::API.open_rest(pr_url)
+          pr.fetch("labels").map { |l| l.fetch("name") }
+        else
+          []
+        end
+
+        runner = random_runner[:name]
+        syntax_job = {
+          name:   "syntax",
+          tap:    tap.name,
+          runner:,
+        }
+
+        matrix = [syntax_job]
+
+        if !syntax_only && !labels&.include?("ci-syntax-only")
+          cask_jobs = if casks&.any?
+            generate_matrix(tap, labels:, cask_names: casks, skip_install:, new_cask:)
+          else
+            generate_matrix(tap, labels:, skip_install:, new_cask:)
+          end
+
+          if cask_jobs.any?
+            # If casks were changed, skip `audit` for whole tap.
+            syntax_job[:skip_audit] = true
+
+            # The syntax job only runs `style` at this point, which should work on Linux.
+            # Running on macOS is currently faster though, since `homebrew/cask` and
+            # `homebrew/core` are already tapped on macOS CI machines.
+            # syntax_job[:runner] = "ubuntu-latest"
+          end
+
+          matrix += cask_jobs
+        end
+
+        syntax_job[:name] += " (#{syntax_job[:runner]})"
+
+        puts JSON.pretty_generate(matrix)
+        github_output = ENV.fetch("GITHUB_OUTPUT", nil)
+        return unless github_output
+
+        File.open(ENV.fetch("GITHUB_OUTPUT"), "a") do |f|
+          f.puts "matrix=#{JSON.generate(matrix)}"
+        end
+      end
+
+      sig { params(cask_content: BasicObject).returns(T.untyped) }
+      def filter_runners(cask_content)
+        # Retrieve arguments from `depends_on macos:`
+        required_macos = case cask_content
+        when /depends_on\s+macos:\s+\[([^\]]+)\]/
+          Regexp.last_match(1).to_s.scan(/\s*(?:"([=<>]=)\s+)?:([^\s",]+)"?,?\s*/).map do |match|
```

Comment:
```suggestion
          T.must(Regexp.last_match(1)).scan(/\s*(?:"([=<>]=)\s+)?:([^\s",]+)"?,?\s*/).map do |match|
```

Bringing this in line with the other change, per the above.

Reply:
Thank you!